search/src/Components/upload/uploader_category_mapper.js

```diff
@@ -0,0 +1,111 @@
+        }
+        this.modifyCategories = this.modifyCategories.bind(this);
+        this.changeCategory = this.changeCategory.bind(this);
+        this.changeSciName = this.changeSciName.bind(this);
+    }
+
+    modifyCategories(index, field, value) {
+        const newCategories = cloneDeep(this.state.categories);
+        newCategories[index][field] = value;
+        this.setState({categories: newCategories});
+    }
+
+    changeCategory(e, index) {
+        this.modifyCategories(index, "category", e.target.value);
+        this.props.handleCategoryStatus(false);
+        this.props.handleErrorMessage("init");
+    }
+
+    changeSciName(e, index) {
+        this.modifyCategories(index, "scitific_name", e.target.value);
+        this.props.handleCategoryStatus(false);
+        this.props.handleErrorMessage("init");
+    }
+
+    render() {
+        const { classes } = this.props;
+        return (
+            <React.Fragment>
+                <p className={classes.summary}>{this.state.categories.length} {this.state.categories.length > 1 ? 'categories' : 'category'} need mapping to weedcoco categories</p>
+                <ol>
+                    {
+                        this.state.categories.map((category, index) => {
+                            return (
+                                <li className={classes.row}>
+                                    <p className={classes.span}>{category.name} is a</p>
+                                    <FormControl className={classes.formControl}>
+                                        <InputLabel id="category">Category</InputLabel>
+                                        <Select
+                                        labelId="category"
+                                        id="category"
+                                        value={category.category}
+                                        onChange={e => {this.changeCategory(e, index)}}
+                                        >
+                                            <MenuItem value={"crop"}>crop</MenuItem>
+                                            <MenuItem value={"weed"}>weed</MenuItem>
+                                        </Select>
+                                    </FormControl>
+                                    <p className={classes.span}>of species</p>
+                                    <TextField label="Scitific name" onChange={e => {e.persist(); this.changeSciName(e, index)}}/>
+                                </li>
+                            )
+                        })
+                    }
+                </ol>
+                <Button variant="contained"
+                        color="primary"
+                        className={classes.applyButton}
+                        onClick={() => {
+                            if (this.state.categories.length > 0 && this.state.categories.length === this.state.categories.filter(category => category.category.length > 0).length) {
```

Comment:
I think this is incorrectly checking `category.category.length`. Shouldn't it be category.scitific_name.length?

Reply:
Is the scientific_name mandatory? I was checking if every entry has a category.